libcanard/canard.h

```diff
@@ -390,8 +420,7 @@ struct CanardInstance
     /// The following API functions may allocate memory:   canardRxAccept(), canardTxPush().
     /// The following API functions may deallocate memory: canardRxAccept(), canardRxSubscribe(), canardRxUnsubscribe().
     /// The exact memory requirement and usage model is specified for each function in its documentation.
-    CanardMemoryAllocate memory_allocate;
-    CanardMemoryFree     memory_free;
+    struct CanardMemoryResource memory;
```

Comment:
For RX it essentially stays the same as it was, just wrapped into `CanardMemoryResource` - but it might change a bit when RX part of the work will be worked on (f.e. usage of narrowed `CanardMemoryDeleter` for the case when only deallocation is expected). Also, I will update list of functions in the above docs according new realities.